xclim/core/calendar.py

```diff
@@ -675,7 +675,14 @@ def percentile_doy(
         (rr.time.dt.year.values, rr.time.dt.dayofyear.values),
         names=("year", "dayofyear"),
     )
-    rrr = rr.assign_coords(time=ind).unstack("time").stack(stack_dim=("year", "window"))
+    if hasattr(xr, "Coordinates"):
+        # xarray > 2023.7.0 will deprecate passing a Pandas MultiIndex directly.
+        # TODO: Remove this condition when pinning xarray above 2023.7.0
```

Comment:
The thing with this one is that the other line _does_ work even with the newest master of xarray! But there's discussions to deprecate it.

I thought that with a "TODO:" it would be easier to find ? But yes indeed, we could try to find all version-dependent code paths at some point and prune what's not needed anymore.

Reply:
I like to add a theme to my todo like `TODO: [version-specific] ....`  or `TODO: [deprecation] ...`
It make it slightly easier to grep and make an inventory.